src/main.rs

```diff
@@ -28,19 +37,45 @@ struct CompilerOptions {
     /// Set the maximum stack size for the current running instance.
     #[clap(short, long, default_value = "10000")]
     stack_size: u32,
+
+    /// Run the compiler in debug mode
+    #[clap(short, long)]
+    debug: bool,
 }
 
 fn main() {
     let opts: CompilerOptions = CompilerOptions::parse();
 
-    println!("Stack_size is {}", opts.stack_size);
+    // print recieved cmdargs, if debug is specified
+    if opts.debug {
+        println!("Stack_size is {}", opts.stack_size);
 
-    for path in opts.includes.into_iter() {
-        println!("Running with {}", path);
+        for path in opts.includes.into_iter() {
+            println!("Running with {}", path);
+        }
     }
 
     match opts.execute {
-        Some(path) => println!("Are we executing -> {}", path),
+        Some(path) => match fs::canonicalize(&path) {
+            Ok(c) => {
+                // Resolve the module path
+                let contents = fs::read_to_string(c.to_str().expect("failed to convert"))
```

Comment:
You don't need the whole `c.to_str().expect("failed to convert")`, `&c` will suffice.